crates/sui-indexer-alt-jsonrpc/src/api/governance.rs

```diff
@@ -41,6 +64,82 @@ impl GovernanceApiServer for Governance {
 
         Ok((rgp as u64).into())
     }
+
+    async fn get_latest_sui_system_state(&self) -> RpcResult<SuiSystemStateSummary> {
+        let Self(ctx) = self;
+
+        /// Fetch the latest version of the object at ID `object_id`, and deserialize its contents
+        /// as a Rust type `T`, assuming that it is a Move object (not a package).
+        ///
+        /// This function extracts the common parts of object loading for this API, but it does not
+        /// generalize beyond that, because:
+        ///
+        /// - It assumes that the objects being loaded are never deleted or wrapped (because it
+        ///   loads using `LatestObjectKey` directly without checking the live object set).
+        ///
+        /// - It first fetches one record from `obj_versions` and then fetches its contents. It is
+        ///   easy to misuse this API to fetch multiple objects in sequence, in a loop, rather than
+        ///   fetching them concurrently.
+        async fn fetch_latest<T: DeserializeOwned>(
+            ctx: &Context,
+            object_id: ObjectID,
+        ) -> Result<T, String> {
+            let id_display = object_id.to_canonical_display(/* with_prefix */ true);
+            let loader = ctx.loader();
+
+            let latest_version = loader
+                .load_one(LatestObjectKey(object_id))
+                .await
+                .map_err(|e| e.to_string())?
+                .ok_or_else(|| format!("Failed to load latest version for {id_display}"))?;
+
+            let stored = loader
+                .load_one(ObjectVersionKey(
+                    object_id,
+                    latest_version.object_version as u64,
+                ))
+                .await
+                .map_err(|e| e.to_string())?
+                .ok_or_else(|| format!("Failed to load object for {id_display}"))?
+                .serialized_object
+                .ok_or_else(|| format!("Failed to load contents for {id_display}"))?;
+
+            let object: Object = bcs::from_bytes(&stored).map_err(|e| e.to_string())?;
+
+            let move_object = object
+                .data
+                .try_as_move()
+                .ok_or_else(|| format!("{id_display} is not a Move object"))?;
+
+            bcs::from_bytes(move_object.contents())
+                .map_err(|e| format!("Failed to deserialize contents for {id_display}: {e}"))
+        }
+
+        let wrapper: SuiSystemStateWrapper = fetch_latest(ctx, SUI_SYSTEM_STATE_OBJECT_ID)
```

Comment:
We have a function `get_sui_system_state(object_store: &dyn ObjectStore)` in `sui-types/sui-system-state` that handles all the wrapper and id derivation churn for you but it does require implementation of `ObjectStore` trait. I suppose you chose not to use that function because you didn't want to implement the trait?

Reply:
Exactly right.